### PR TITLE
[Tokenizer] Unify tokenizer _pad

### PR DESCRIPTION
**Pull request by DrownFish19**
### PR types

Function optimization
### PR changes

APIs
### Description

Unify tokenizer _pad function.

* Move the `attention_mask([1,seq_len,seql_len])` padding action into tokenizer_base `_pad`.
* Move `attn_mask_startend_row_indices` padding action into tokenizer_base `_pad`.

 [[FlashMask] Add FlashMask for Qwen2 #9264](https://github.com/PaddlePaddle/PaddleNLP/pull/9264)误差范围验证基于此PR

### CODE REVIEW

**Review by lugimzzz — DISMISSED** [review]
LGTM

**Review by ZHUI — APPROVED** [review]
